### DISCUSSION TITLE
Full software backup vs genmon backup

### DISCUSSION THREAD

**Discussion by DavePanetta** [jgyates/genmon#1060] [Questions] [answered]
I have been slowly building my genmon software, and having been out of touch with linux/unix and anything else for that matter, for over 20 years I have  crashed and burned a few times. I am getting pretty good at rebuilding everything, but getting a little tired of it.

My question is, is there a better way than doing a whole rebuild?
And then using my copies of the genmon backup to get back to where I was?

I have read several things about cloning the SD card, but they all seem to have problems.
On a running PI it is suggested to use dd, but I wonder about coping a moving tartget so to speak.

I am running my pi headless and do not have a keyboard and monitor for it.
I do have VNC enabled and have had minor success using it. I also use putty mostly without problems.
A windows solution would be great if it isn't very drawn out.

It seems like a fairly simple need, but Microsoft seems to not want us to do it at least easily.

Does anyone have a suggestion on how to get a complete SD card clone without going through as much work as just rebuilding it.

Thanks for any help,
Dave

**Comment by jgyates** [comment]
This thread has some discussion on this topic: https://github.com/jgyates/genmon/discussions/1057

**Reply by DavePanetta** [reply]
1057 has good info I will do some of the things suggested.
My latest problems make me wonder if there is also a RAM memory test. For some reason Something happened and I started getting a message about I/O errors after trying sudo --- something  - don't remember. and then bash complaining that the command was not recognized. Thereafter everything got the same kind of response.
I had to power cycle it to reboot (bad) but it seemed to be ok for a while and then it started doing it again.
I am about to rebuild, but was hoping for a way to clone the card before I did something to crash it again.

**Reply by skipfire** [reply]
Commands not found aren't a concern, it's just a missing package and not a hardware issue.  For the I/O error, that would depend on the command as it could be a hardware config issue.  Memory issues rarely result in readable error messages.  If you want to test memory, I pulled from https://forums.raspberrypi.com/viewtopic.php?f=63&t=103847#p717845.

```
sudo apt-get update && sudo apt-get install memtester
sudo memtester 512 10
```
512 memory to test
10 is number of loops

**Reply by DavePanetta** [reply, marked as answer]
Thanks,
 I was gone for a bit. You didn't understand. Once I got that error, any command got that error and I could not do a software shutdown or reboot. - So rebuild. I just hope it was corrupt SW and not my HW.

**Comment by DavePanetta** [comment]
As I said in the forum, I was away for a bit (had to take my grandkids
home). Thanks for all the help.

I am not sure what is going on with my pi, if I restart it manually by
power cycle it comes up ok for a little while, then for no
particular reason it just starts getting that weird error and I can only do
a manual power down.
So the next thing is a complete re build. I'll try bookworm again. I had
problems the other day, maybe I screwed something up.
so I will try again.
Thanks

**Reply by DavePanetta** [reply]
JUST an FYI.
https://www.helping.ninja/how-to-migrate-raspberry-pi-sd-card-to-a-usb-ssd-speedtest/

I found this article interesting. I seemed to work with one of my pi 4 systems using bullseye 64 bit. 
I didn't actually use a thumb drive, I used a 64 GB micro SD card (formatted Fat32) in my USB adapter.
The speed test results were only limited improvement, but the PI 4 did boot from the card in the USB adapter
after removing the SD card from the reader
I am not sure if this is worth much other than a way to clone a micro SD card.  Might be an OK way to add a boot able SSD drive.
Any comments?

**Comment by Jkh62** [comment]
I used Win32Diskimager and made a direct image from the working SD card on my PC.
Loading a replacement card might not take very long but having a plug in ready to go is nice to have.

I never could figure out if you can get a clean image from a running pi without corrupting or crashing it.
